language/move-prover/bytecode/src/compositional_analysis.rs

```diff
@@ -58,19 +58,21 @@ impl<'a> SummaryCache<'a> {
 
 /// Trait that lifts an intraprocedural analysis into a bottom-up, compositional interprocedural
 /// analysis. The derived `summarize` function computes a postcondition for the procedure under
-/// analysis, then stores it in the summary cache for other procedures to use.
-// TODO: allow client analyses to customize this by providing a function that converts a
-// postcondition into a summary
+/// analysis, then stores it in the summary cache for other procedures to use. The function
+/// 'summarize_map' is the same as 'summarize' but with the additional argument 'map' that is
+/// applied to the postcondition; the result of this function application is stored in the summary
+/// cache in place of the computed postcondition.
 pub trait CompositionalAnalysis: DataflowAnalysis
 where
     Self::State: 'static,
 {
-    fn summarize(
+    fn summarize_map<T:std::any::Any>(
```

Comment:
- What is the difference between `T: std::any::Any` and plain `T`?
- It might makes sense to constrain `T` by `AbstractDomain` here. If we ever get around to handling recursion in the compositional analysis framework, I think it will need to be able to join summaries.

Reply:
`std::any::Any` is probably not desired here (https://doc.rust-lang.org/std/any/index.html). I used it because it matched the type of annotations. The any trait is used to enable dynamic type checking which we are not doing in this function. I will change `T` to `'static`.

I will add `AbstractDomain` as a trait here as well for the reason you listed.